### PR TITLE
feat(clientConnector): support for default client connector

### PR DESCRIPTION
**Pull request by eperjust**
update to support default client connector config. This will allow consumers to implement the default client connector method to add support for proxy when needed.

### CODE REVIEW

File: client/client-samples/sample-client-reactive-football/src/test/java/io/americanexpress/sample/client/reactive/football/config/FootballReactiveClientTestConfig.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * {@code FootballReactiveClientTestConfig} contains configurations for tests.
+ *
+ * @author eperjust
+ */
+@Configuration
+@Import({FootballReactiveClientConfig.class, UtilitiesCommonConfig.class})
+public class FootballReactiveClientTestConfig {
+}
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/test/java/io/americanexpress/sample/client/reactive/football/config/FootballReactiveClientTestConfig.java]: add extra line at end of file

File: client/client-samples/sample-client-reactive-football/src/main/resources/client-football.properties [hunk 2]
```diff
@@ -0,0 +1,3 @@
+client.url=https://api.squiggle.com.au/?q=teams
+client.proxy.host=proxy.aexp.com
+client.proxy.port=8080
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/resources/client-football.properties]: add extra line at end of file

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootballResponse.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+import io.americanexpress.synapse.client.rest.model.BaseClientResponse;
+
+/**
+ * {@code FootballResponse} is the response object for football client.
+ *
+ * @author eperjust
+ */
+public class FootballResponse implements BaseClientResponse {
+}
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootballResponse.java]: add extra line at end of file

File: client/client-samples/sample-client-reactive-football/src/main/resources/client-football.properties [hunk 4]
```diff
@@ -0,0 +1,3 @@
+client.url=https://api.squiggle.com.au/?q=teams
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/resources/client-football.properties]: add copy writes

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootBallRequest.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+import io.americanexpress.synapse.client.rest.model.BaseClientRequest;
+
+/**
+ * {@code FootBallRequest} is the request object for football client.
+ *
+ * @author eperjust
+ */
+public class FootBallRequest implements BaseClientRequest {
+}
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootBallRequest.java]: add extra line at end of file

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/handler/FootballReactiveResponseErrorHandler.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.stereotype.Component;
+
+/**
+ * {@code FootballReactiveResponseErrorHandler} handles errors for reactive football client.
+ *
+ * @author eperjust
+ */
+@Component
+public class FootballReactiveResponseErrorHandler extends BaseReactiveRestResponseErrorHandler {
+}
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/handler/FootballReactiveResponseErrorHandler.java]: add extra line at end of file

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/config/FootballReactiveClientConfig.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    @Override
+    protected ReactorClientHttpConnector defaultClientConnector() {
+        HttpClient httpClient = HttpClient.create()
+                .proxy(proxy -> proxy.type(ProxyProvider.Proxy.HTTP)
+                        .host(this.clientProxyHost)
+                        .port(this.clientProxyPort));
+        return new ReactorClientHttpConnector(httpClient);
+    }
+}
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/config/FootballReactiveClientConfig.java]: add extra line at end of file

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/client/FootballReactiveClient.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    /**
+     * Argument constructor creates a new instance of BaseReactiveRestClient with given values.
+     *
+     * @param httpHeadersFactory               HTTP headers factory used to produce the custom HTTP headers required to consume the back end service
+     * @param reactiveRestResponseErrorHandler used to handle errors from the reactive REST client
+     */
+    protected FootballReactiveClient(FootballReactiveClientHeadersFactory httpHeadersFactory, BaseReactiveRestResponseErrorHandler reactiveRestResponseErrorHandler) {
+        super(httpHeadersFactory, reactiveRestResponseErrorHandler);
+    }
+}
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/client/FootballReactiveClient.java]: add extra line at end of file

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootBallRequest.java [hunk 9]
```diff
@@ -0,0 +1,11 @@
+package io.americanexpress.sample.client.reactive.football.model;
+
+import io.americanexpress.synapse.client.rest.model.BaseClientRequest;
+
+/**
+ * {@code FootBallRequest} is the request object for football client.
+ *
+ * @author eperjust
+ */
+public class FootBallRequest implements BaseClientRequest {
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootBallRequest.java]:
```suggestion
public class FootBallClientRequest implements BaseClientRequest {
```

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootballResponse.java [hunk 10]
```diff
@@ -0,0 +1,11 @@
+package io.americanexpress.sample.client.reactive.football.model;
+
+import io.americanexpress.synapse.client.rest.model.BaseClientResponse;
+
+/**
+ * {@code FootballResponse} is the response object for football client.
+ *
+ * @author eperjust
+ */
+public class FootballResponse implements BaseClientResponse {
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootballResponse.java]:
```suggestion
public class FootballClientResponse implements BaseClientResponse {
```

File: client/client-samples/sample-client-reactive-football/src/main/resources/client-football.properties [hunk 11]
```diff
@@ -0,0 +1,3 @@
...
+client.proxy.host=proxy.aexp.com
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/resources/client-football.properties]: change this please!

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/client/FootballReactiveClientHeadersFactory.java [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+
+/**
+ * {@code FootballReactiveClientHeadersFactory} creates headers for the football reactive client.
+ *
+ * @author eperjust
+ */
+@Component
+public class FootballReactiveClientHeadersFactory extends BaseClientHttpHeadersFactory<FootBallRequest> {
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/client/FootballReactiveClientHeadersFactory.java]: move this class to a factory package

**Review by guttff — APPROVED** [review]

File: client/client-samples/sample-client-reactive-football/src/test/java/io/americanexpress/sample/client/reactive/football/config/FootballReactiveClientTestConfig.java (hunk 1 above)
Comment (eperjust) [reply client/client-samples/sample-client-reactive-football/src/test/java/io/americanexpress/sample/client/reactive/football/config/FootballReactiveClientTestConfig.java]: done

File: client/client-samples/sample-client-reactive-football/src/main/resources/client-football.properties (hunk 2 above)
Comment (eperjust) [reply client/client-samples/sample-client-reactive-football/src/main/resources/client-football.properties]: done

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootballResponse.java (hunk 3 above)
Comment (eperjust) [reply client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootballResponse.java]: done

File: client/client-samples/sample-client-reactive-football/src/main/resources/client-football.properties (hunk 4 above)
Comment (eperjust) [reply client/client-samples/sample-client-reactive-football/src/main/resources/client-football.properties]: done

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootBallRequest.java (hunk 5 above)
Comment (eperjust) [reply client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootBallRequest.java]: done

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/client/FootballReactiveClient.java (hunk 8 above)
Comment (eperjust) [reply client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/client/FootballReactiveClient.java]: done

File: client/client-samples/sample-client-reactive-football/src/main/resources/client-football.properties (hunk 11 above)
Comment (eperjust) [reply client/client-samples/sample-client-reactive-football/src/main/resources/client-football.properties]: done

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootBallRequest.java (hunk 9 above)
Comment (eperjust) [reply client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootBallRequest.java]: done

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/client/FootballReactiveClientHeadersFactory.java (hunk 12 above)
Comment (eperjust) [reply client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/client/FootballReactiveClientHeadersFactory.java]: done

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootballResponse.java (hunk 10 above)
Comment (eperjust) [reply client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/model/FootballResponse.java]: done

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/config/FootballReactiveClientConfig.java [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+ */
+@Configuration
+@ComponentScan("io.americanexpress.sample.client.reactive.football")
+@PropertySource("classpath:client-football.properties")
+public class FootballReactiveClientConfig extends BaseReactiveRestClientConfig {
+
+    /**
+     * The client proxy host information.
+     */
+    @Value("${client.proxy.host}")
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/config/FootballReactiveClientConfig.java]: user the environment

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/config/FootballReactiveClientConfig.java [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+    private String clientProxyHost;
+
+    /**
+     * The client proxy port information.
+     */
+    @Value("${client.proxy.port}")
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/config/FootballReactiveClientConfig.java]: user the environment

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/config/FootballReactiveClientConfig.java [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+    private int clientProxyPort;
+
+    /**
+     * Reactive client for making call to weather api.
+     */
+    private final FootballReactiveClient footballReactiveClient;
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/config/FootballReactiveClientConfig.java]:
```suggestion
    private final FootballGetReactiveRestClient footballGetReactiveClient;
```

File: client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/config/FootballReactiveClientConfig.java [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+        this.footballReactiveClient = footballReactiveClient;
+        this.footballReactiveResponseErrorHandler = footballReactiveResponseErrorHandler;
+    }
+
+    /**
+     * This will initialize the default webclient
+     *
+     * @param destinationUrl of the provider
+     */
+    @Value("${client.url}")
```
Comment (guttff) [inline client/client-samples/sample-client-reactive-football/src/main/java/io/americanexpress/sample/client/reactive/football/config/FootballReactiveClientConfig.java]: make more specific

**Pull request merged**